libp2p/src/builder/phase/build.rs

```diff
@@ -0,0 +1,29 @@
+use super::*;
+use crate::SwarmBuilder;
+use libp2p_core::Transport;
+use libp2p_swarm::Swarm;
+
+pub struct BuildPhase<T, B> {
+    pub(crate) behaviour: B,
+    pub(crate) transport: T,
+    pub(crate) swarm_config: libp2p_swarm::Config,
+}
+
+const CONNECTION_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
```

Comment:
This is currently not configurable, right? Should we add it to `swarm::Config`?

Reply:
I would do so once we have a user for it. I would guess that `10s` covers 99% of all cases. Can you think of a use-case that would require the config option today already?

Reply:
Fair enough!